sandbox/src/main/java/org/robolectric/internal/bytecode/ShadowWrangler.java

```diff
@@ -181,7 +181,7 @@ public MethodHandle findShadowMethodHandle(
               MethodType actualType = isStatic ? methodType : methodType.dropParameterTypes(0, 1);
               Class<?>[] paramTypes = actualType.parameterArray();
 
-              Method shadowMethod = pickShadowMethod(definingClass, name, paramTypes);
+              Method shadowMethod = pickShadowMethod(definingClass, name, actualType);
```

Comment:
Can we just keep ShadowWrangler as-is for now instead of checking the `@ClassName` return type annotation?

We don't check the return type when checking any other shadow methods, so I am not seeing a strong reason to special-case the `@ClassName` return types. It adds a lot of complexity to ShadowWrangler but it does not seem to make a difference functionally. As far as I know, when doing invokedynamic binding, the return type is not considered.

`validator/SdkStore.java` is what cares about the return type, so we should just add logic to that to check it.

Reply:
Hi Michael,
I have a side question, may I ask when the SdkStore logic is executed? 1) in the stage of `pre-instrument Android-all jars to publish to maven` or 2) when in the early stage of unit-test startup during which verify all shadows are correctly defined or 3) in early stage of each test function to verify that shadows used in current test-function are correctly defined? or 4) any other timing?

Reply:
done. 
make SdkStore take care of ClassName annotation in function return type

Reply:
Hello reviewers,
by running test `./gradlew :shadows:framework:compileJava` locally, I  found that 
1, ClassName must target to `{ElementType.TYPE_USE, ElementType.PARAMETER}`, otherwise the annotation will not able be found in SdkStore logic.
2, sometimes I have to enumerate all annotations for a parameter/return type to find ClassName annotation, sometimes I can use xxx.getAnnotation(ClassName.class) to quickly get handle to the annotation

Reply:
Hi @Mia0451 ,

We can start off using TYPE_USE for now, and perhaps as a follow-up we can clean it up.

Thanks!

Reply:
got it